src/R/Components/Impl/History/Implementation/RHistoryProvider.cs

```diff
@@ -16,18 +16,29 @@ namespace Microsoft.R.Components.History.Implementation {
     [Export(typeof(IRHistoryProvider))]
     internal class RHistoryProvider : IRHistoryProvider {
         private readonly ITextBufferFactoryService _textBufferFactory;
-        private readonly IFileSystem _fileSystem;
         private readonly IEditorOperationsFactoryService _editorOperationsFactory;
         private readonly IRtfBuilderService _rtfBuilderService;
         private readonly ITextSearchService2 _textSearchService;
         private readonly IRSettings _settings;
         private readonly IContentTypeRegistryService _contentTypeRegistryService;
         private readonly Dictionary<ITextBuffer, IRHistory> _histories;
 
+        private IFileSystem _fileSystem;
+        internal IFileSystem FileSystem {
```

Comment:
AFAIK, in RHistory tests, IFileSystem implementation is provided through MEF, so you can leave it in ctor but mark parameter with AllowDefault

---
In reply to: [68614367](https://github.com/Microsoft/RTVS/pull/2004#discussion_r68614367) [](ancestors = 68614367,68613538)